lib/mysql.js

```diff
@@ -200,17 +200,8 @@ MySQL.prototype.executeSQL = function(sql, params, options, callback) {
   }
 };
 
-/**
- * Update if the model instance exists with the same id or create a new instance
- *
- * @param {String} model The model name
- * @param {Object} data The model instance data
- * @param {Function} [callback] The callback function
- */
-MySQL.prototype.updateOrCreate = MySQL.prototype.save =
-  function(model, data, options, callback) {
-    var fields = this.buildFields(model, data);
-
+ MySQL.prototype.modifyOrCreate = function(model, data, options, buildFields ,callback) {
+    var fields = buildFields(model, data);
```

Comment:
In fact, you can probably leave updateOrCreate alone. Then add a options.replace and check if (options.replace), then use this.buildReplaceFields instead of buildFields.